x-pack/plugins/lens/public/editor_frame_service/error_helper.tsx

```diff
@@ -144,9 +186,16 @@ export function getOriginalRequestErrorMessages(error?: ExpressionRenderError |
       }
     }
   } else if (error?.message) {
-    errorMessages.push(error?.message);
+    errorMessages.push(error.message);
   }
-  return errorMessages;
+  return errorMessages.map((message) => ({
+    uniqueId: typeof message === 'string' ? message : message.short,
+    severity: 'error',
+    displayLocations: [{ id: 'visualizationOnEmbeddable' }],
```

Comment:
Does it make sense to also display this on the offending dimension?

Reply:
Never mind—missed that this is just on the embeddable.

Reply:
It's not just for embeddable, but the main problem is to map back the error to a specific dimension as the error is extracted from a network response, so runtime. The dimension should already show a validation error at configuration time, validation message coming from the operation definition itself.